file.go

```diff
@@ -1,37 +1,127 @@
 package jcsv
 
-import "os"
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	jsonData []uint8
+	csvData [][]string
+	csvHeader []string
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	var fileObj file
+	fileStream,fileOpenError := os.Open(path)
+	if fileOpenError!=nil{
+		return file{},fileOpenError
+	}
+	var fileReadError error
+	fileObj.jsonData,fileReadError=ioutil.ReadAll(fileStream)
+	fileStream.Close()
+	if fileReadError!=nil {
+		return file{},fileReadError
+	}
+	return fileObj,nil
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var fileRead file
+	var fileReadError error
+	fileRead.jsonData,fileReadError=ioutil.ReadAll(f)
+	if fileReadError!=nil {
+		return file{},fileReadError
+	}
+	return fileRead,nil
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	var returnFile file
+	csvFile, err := os.Open(path)
+	if err != nil {
+		fmt.Println(err)
+		return file{},err
+	}
+	readBuffer,readError := csv.NewReader(csvFile).ReadAll()
+	if readError!=nil{
+		return file{},readError
+	}
+	var i int
+	if hasHeaders {
+		for j := 0; j < len(readBuffer[0]); j++ {
+			returnFile.csvHeader = append(returnFile.csvHeader, readBuffer[0][j])
+		}
+		i=1
+	}else {
```

Comment:
Code is not formatted.

Reply:
If you are talking about '}' followed by "else", GO's compiler prompts syntax error on this.

Reply:
I tried move '{' to next line, and when I save code in VS code, it automatically moves the '{' (that follows "else") on the previous line.